### DISCUSSION TITLE
Extending Warg CLI Features vs Creating Other CLIs

### DISCUSSION THREAD

**Discussion by calvinrp** [bytecodealliance/registry#217] [General]
### Do we want to encourage or discourage alternative Warg protocol compatible CLIs?

From casual discussions, there is interest in platform-specific CLI tools integrating Warg support. Is this something that we want to make easier to do?

Should the Warg CLI support and standardize authentication for private registries?

We plan on revisiting key management alternatives. However, there is interest in using a service for key signing. Should this feature be implemented in alternative Warg-compatible CLI?

**Comment by lann** [comment]
If we are planning to make "assisted signing" a Warg feature we should probably implement it in the reference client.

That said, there is nothing wrong with alternative tooling with the usual caveat that we haven't really stabilized (Rust or protocol) APIs.

**Reply by calvinrp** [reply]
Should private registry authentication be in scope for the reference client CLI?

**Reply by lann** [reply]
It certainly could be, as long as it uses something standardized like an oauth code flow.

**Reply by esoterra** [reply]
It should definitely be in scope for the reference client library because that will be used by tools like Cargo Component and while it might be practical to have your own CLI if you need one, it won't be practical to make your own Cargo Component.
